src/libvirtApi/snapshot.js

```diff
@@ -39,78 +39,55 @@ export function snapshotCurrent({ connectionName, objPath }) {
+            return { xml, isCurrent };
+        }));
 
-                // WA to avoid Promise.all() fail-fast behavior
-                const toResultObject = (promise) => {
-                    return promise
-                            .then(result => ({ success: true, result }))
-                            .catch(error => ({ success: false, error }));
-                };
-
-                return Promise.all(promises.map(toResultObject))
-                        .then(snapXmlList => {
-                            snapXmlList.forEach(snap => {
-                                if (snap.success) {
-                                    const result = snap.result;
-                                    const snapParams = parseDomainSnapshotDumpxml(result.xml[0]);
-                                    snapParams.isCurrent = result.isCurrent[0];
-                                    snaps.push(snapParams);
-                                } else {
-                                    console.warn("DomainSnapshot method GetXMLDesc failed", snap.error.toString());
-                                }
-                            });
-                            return store.dispatch(updateDomainSnapshots({
-                                connectionName,
-                                domainPath,
-                                snaps: snaps.sort((a, b) => a.creationTime - b.creationTime)
-                            }));
-                        });
-            })
-            .catch(ex => {
-                if (ex.name === 'org.freedesktop.DBus.Error.UnknownMethod')
-                    logDebug("LIST_DOMAIN_SNAPSHOTS action failed for domain", domainPath, ", not supported by libvirt-dbus");
-                else
-                    console.warn("LIST_DOMAIN_SNAPSHOTS action failed for domain", domainPath, ":", JSON.stringify(ex), "name:", ex.name);
-                return store.dispatch(updateDomainSnapshots({
-                    connectionName,
-                    domainPath,
-                    snaps: -1
-                }));
-            });
+        const snaps = [];
+        snapXmlList.forEach(snap => {
+            if (snap.status === 'fulfilled') {
+                const snapParams = parseDomainSnapshotDumpxml(snap.value.xml);
+                snapParams.isCurrent = snap.value.isCurrent;
+                snaps.push(snapParams);
+            } else {
+                console.warn("DomainSnapshot method GetXMLDesc failed", snap.reason.toString());
+            }
+        });
+        store.dispatch(updateDomainSnapshots({
+            connectionName,
+            domainPath,
+            snaps: snaps.sort((a, b) => a.creationTime - b.creationTime)
+        }));
+    } catch (ex) {
+        if (ex.name === 'org.freedesktop.DBus.Error.UnknownMethod')
+            logDebug("LIST_DOMAIN_SNAPSHOTS action failed for domain", domainPath, ", not supported by libvirt-dbus");
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1400-20240125-053137-bfbdf98b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)